### PR TITLE
Make client-go authoritative

### PR DESCRIPTION
**Pull request by caesarxuchao**
client-go now is authoritative except for `client-go/pkg/version`, which I think we can leave as is for a while.

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -66,27 +66,26 @@ ${clientgen} --output-base "${KUBE_ROOT}/vendor" --clientset-path="k8s.io/client
...
   cd ${KUBE_ROOT}
-  # because client-gen doesn't do policy/v1alpha1, we have to skip it too
-  find pkg/apis -name types.go | xargs -n1 dirname | sort | grep -v pkg.apis.policy.v1alpha1
+  find pkg/apis -maxdepth 2 -name types.go | xargs -n1 dirname | sort 
 )
 )
-listergen_kubernetes_apis=(${listergen_kubernetes_apis[@]/#/k8s.io/kubernetes/})
-listergen_staging_apis=(
+listergen_internal_apis=(${listergen_internal_apis[@]/#/k8s.io/kubernetes/})
+LISTERGEN_INTERNAL_APIS=$(IFS=,; echo "${listergen_internal_apis[*]}")
```
Comment (sttts) [inline hack/update-codegen.sh]: strange var name reuse. How about listergen_internal_apis_csv?

File: hack/update-codegen.sh [hunk 2]
```diff
@@ -87,36 +87,40 @@ $(
...
   cd ${KUBE_ROOT}
-  # because client-gen doesn't do policy/v1alpha1, we have to skip it too
-  find pkg/apis -name types.go | xargs -n1 dirname | sort | grep -v pkg.apis.policy.v1alpha1
+  find pkg/apis -maxdepth 2 -name types.go | xargs -n1 dirname | sort 
 )
 )
-informergen_kubernetes_apis=(${informergen_kubernetes_apis[@]/#/k8s.io/kubernetes/})
-informergen_staging_apis=(
+informergen_internal_apis=(${informergen_internal_apis[@]/#/k8s.io/kubernetes/})
+INFORMERGEN_INTERNAL_APIS=$(IFS=,; echo "${informergen_internal_apis[*]}")
```
Comment (sttts) [inline hack/update-codegen.sh]: as before: informergen_internal_apis_csv

File: staging/src/k8s.io/kube-gen/cmd/informer-gen/generators/customargs.go [hunk 3]
```diff
@@ -22,10 +22,12 @@ type CustomArgs struct {
...
 	InternalClientSetPackage  string
 	ListersPackage            string
+	SingleDirectory           bool
 }
 
 func (ca *CustomArgs) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&ca.InternalClientSetPackage, "internal-clientset-package", ca.InternalClientSetPackage, "the full package name for the internal clientset to use")
 	fs.StringVar(&ca.VersionedClientSetPackage, "versioned-clientset-package", ca.VersionedClientSetPackage, "the full package name for the versioned clientset to use")
 	fs.StringVar(&ca.ListersPackage, "listers-package", ca.ListersPackage, "the full package name for the listers to use")
+	fs.BoolVar(&ca.SingleDirectory, "single-directory", ca.SingleDirectory, "if false, will output to \"internalversion\" and \"externalversion\" subdirecotries")
```
Comment (sttts) [inline staging/src/k8s.io/kube-gen/cmd/informer-gen/generators/customargs.go]: subdirectories

Comment (sttts) [inline staging/src/k8s.io/kube-gen/cmd/informer-gen/generators/customargs.go]: Better: if true, omit the intermediate \"internalversion\" and \"externalversion\" subdirs

File: hack/update-staging-client-go.sh [hunk 4]
```diff
@@ -1,36 +0,0 @@
-#!/bin/bash
```
Comment (sttts) [inline hack/update-staging-client-go.sh]: 🎉

File: hack/verify-staging-client-go.sh [hunk 5]
```diff
@@ -1,29 +0,0 @@
-#!/bin/bash
```
Comment (sttts) [inline hack/verify-staging-client-go.sh]: 🍺

File: staging/copy.sh [hunk 6]
```diff
@@ -1,198 +0,0 @@
-#!/bin/bash
```
Comment (sttts) [inline staging/copy.sh]: 🚀

**Comment by sttts** [comment]
some nits. Overall, very good!

**Comment by deads2k** [comment]
very happy to see it.

/approve

**Comment by caesarxuchao** [comment]
Thanks for the reviews. Comments addressed and squashed a bit. And thank you @jcbsmpsn, your https://github.com/kubernetes/kubernetes/pull/48231 was the big step.

**Comment by caesarxuchao** [comment]
Added a new commit that updates staging/README.md.

**Comment by caesarxuchao** [comment]
/assign @lavalamp 
for approval

**Pull request merged**